storage/federatedx/ha_federatedx.cc

```diff
@@ -3618,6 +3627,11 @@ int ha_federatedx::discover_assisted(handlerton *hton, THD* thd,
     goto err2;
 
   query.copy(rdata[1], rlen[1], cs);
+  cut_offset= query.strstr(cut_clause, 17); // ignore CREATE TABLE, spaces, identifier
+  if (cut_offset > -1)
+  {
+    query.replace(cut_offset, cut_clause.length(), NULL, 0);
+  }
   query.append(STRING_WITH_LEN(" CONNECTION='"), cs);
```

Comment:
To avoid false positives (`create table t1 (a varchar(200) default ' WITH SYSTEM VERSIONING')`) let's only detect this at the very end of the query. Otherwise looks ok.

Eventually federated will need to generate table definition itself, like other engines are doing.

Reply:
Fixed by separate commit (for easier review). If everything is ok, please squash it to previous.